libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -66,6 +66,7 @@ class NavigationView @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     accessToken: String = attrs.navigationViewAccessToken(context),
+    viewModelStoreOwner: ViewModelStoreOwner = context.toViewModelStoreOwner()
```

Comment:
```suggestion
    hostingFragment: Fragment? = null
```

The fragment is both, a lifecycle owner and a view model store owner. IMO you'd want to just inject it directly and make it clear what is being supported. We could convert it into another constructor.

Why would you want to use `ViewLifecycleRegistry` when you're using a `Fragment`?

Reply:
When prototyping, left some ideas over here https://github.com/mapbox/mapbox-navigation-android/pull/5820

resolving this thread